railties/lib/rails/generators/app_base.rb

```diff
@@ -469,6 +469,26 @@ def run_css
         end
       end
 
+      def report_dependency_issues
+        dependencies = {
+          active_storage: :active_record,
+          action_mailbox: :active_storage,
+          action_text: :active_storage,
+          hotwire: :javascript
+        }
+
+        dependencies.each do |component, requirement|
+          next if options["skip_#{component}"]
+          while requirement
+            if options["skip_#{requirement}"]
+              say "#{component} was not installed because it requires a skipped component (#{requirement}).", :red
+              break
+            end
+            requirement = dependencies[requirement]
+          end
+        end
+      end
```

Comment:
I'm thinking something like this:

```ruby
def confirm_implied_options
  option_implicators = {
    skip_active_storage: [:skip_active_record],
    skip_action_mailbox: [:skip_active_storage],
    skip_action_text: [:skip_active_storage],
    skip_hotwire: [:skip_javascript],
  }

  # TODO Potentially add recursive resolution, like in your code.
  implied_options = option_implicators.transform_values do |implicators|
    implicators.select { |implicator| options[implicator] }
  end.compact_blank
  # TODO Add `require "active_support/core_ext/enumerable"` for `compact_blank`.

  if !implied_options.empty?
    say "Based on the specified options, the following options will also be activated:"

    implied_options.each do |implied, implicators|
      due_to = implicators.map { |implicator| "--#{implicator.to_s.tr("_", "-")}" }.join(" ")
      say "* --#{implied.to_s.tr("_", "-")} (due to: #{due_to})"
    end

    if yes?("Proceed?")
      self.options = options.merge(implied_options.transform_values { true }).freeze
    else
      exit # TODO Is there a more Thor-idiomatic way of exiting / aborting?
    end
  end
end
```

That is just a sketch, so there may be errors.  Also, if we merge the implied options (as above), we can remove the redundant checks from the affected tasks (e.g. `options[:skip_javascript]` from `hotwire_gemfile_entry`).

Reply:
@jonathanhefner Ok, skipped implied options are now flagged and, if the user chooses to proceed, activated:
![Screen Shot 2022-06-01 at 2 07 56 AM](https://user-images.githubusercontent.com/76491823/171340520-5a023fc7-21f3-4663-b277-54b07f44cb85.png)

This isn't ready to merge yet: I still need to remove redundant checks and fix the 18 tests that now break because components are being installed even when a prereq is initially skipped--but I pushed the basic code so I could get confirmation that this is the preferred functionality. (Suggestions for improving the code are also welcome!)  Also, I'm not very familiar with this testing suite: how can I mock `yes?` so that tests will run without receiving user input in response to the "Proceed?"?